packages/angular/cli/lib/config/workspace-schema.json

```diff
@@ -106,6 +106,15 @@
               "type": "string"
             }
           }
+        },
+        "completion": {
```

Comment:
`completion` is only valid in global context. This need some more thought.

Reply:
@alan-agius4 -- I gave this another look based on your feedback. Not sure if this is what you had in mind, but I added a check inside `ConfigCommandModule`'s `set()` method that will throw an "Invalid Path" error if a user tries to set `cli.completion.prompted` without the `--global` option.

Reply:
@t-bowersox, thanks for taking a look at this. 

I noticed that we have an unused `global` definition in the schema. 

https://github.com/angular/angular-cli/blob/9fafb2e1251f6790392a1f9c84854086c85d2191/packages/angular/cli/lib/config/workspace-schema.json#L551-L568

What we can do is to use this definition and validate against that. 

I'll summarize what needs to be done, but don't feel obligated to take this. I can take over if you don't wish to continue working on this.

- update the the `cli` property to include the `completion` option. This can be done using `anyOf`.
- Remove `format` from `$schema` property.
- Revert changes in https://github.com/angular/angular-cli/pull/23404/files#diff-2196c0c0d0c3badafa233d0e2e43d633e8059652f1021f4c1605f025cccfd6e8 and remove all logic related to `validGlobalCliPaths`.
- Update [validateWorkspace](https://github.com/angular/angular-cli/blob/9fafb2e1251f6790392a1f9c84854086c85d2191/packages/angular/cli/src/utilities/config.ts#L243-L253) to validate against the global definitions, something like.

```ts
export async function validateWorkspace(data: json.JsonObject, isGlobal: boolean): Promise<void> {
  const schema = readAndParseJson(workspaceSchemaPath);
  // We should eventually have a dedicate global config schema and use that to validate.
  const schemaToValidate = isGlobal
    ? {
        oneOf: [{ '$ref': '#/definitions/global' }],
        definitions: schema['definitions'],
      }
    : schema;
```
- Add an E2E test in [tests/legacy-cli/e2e/tests/commands/config](https://github.com/angular/angular-cli/blob/main/tests/legacy-cli/e2e/tests/commands/config)